docs/zkevm/l1-info-tree.md

```diff
@@ -0,0 +1,53 @@
+# L1 Info Tree
+
+## Summary from etrog slide deck
+
+Link: https://docs.google.com/presentation/d/1QvmaCwDIpCeMlU-KkOfCEVtekuKP7_GJEJJ2wSY9syU/edit#slide=id.g282f6c192a8_1_34
+
+- Data structure present in the Smart Contracts to allow:
+  - Granularity in GlobalExitRoot during batch processing
+  - Add L1 information to L2
+
+- Information included in the L1 Info Tree
+  - GlobalExitRoot
+  - L1 block hash
+  - minTimestamp
+
+![diagram.png](diagram.png)
+
+- L1InfoTree has 32 static levels
+- Incremental
+  - Add new leaf when a new GER is computed in the SC
+- Given an index → could always get (verify) a L1Data
+  - If an index does not still contain any L1Data → L1Data = 0
+- Only save the latest l1InfoRoot in the SC
+- Index 0 has GER = 0 (Special index)
+  - Allow not changing GER with index = 0
+  - Less gas cost data-availability
+
+## Sequencer concerns
+
+When building batches the sequencer is responsible for tracking the L1 info tree and storing information about the indexes, and tracking which indexes have already been used during block creation.
+
+The index is known by simply tracking the events from the GER manager contract on the L1 from the start of the network and incrementing by 1 each time a new event is found.  Index 0 is a special index and is used to denote no change.
+
+At the point of building a new block the sequencer will check to see if the next index is available to use:
+
+- Current index + 1 exists in DB?
```

Comment:
This means that the `l1InfoTreeIndex` is finalized on L1 I guess.
Therefore, no future `l1InfoTreeIndex` would be used

Maybe add a note tho explain above comment